src/backend/metal/src/lib.rs

```diff
@@ -61,11 +77,9 @@ impl Instance {
         Instance {}
     }
 
-    pub fn create_surface(&self, window: &winit::Window) -> Surface {
+    pub fn create_surface(&self, window: &NativeWindow) -> Surface {
```

Comment:
I don't think having a trait here is really necessary. We don't need to be generic here. Instead, we could have:
```rust
pub fn create_surface_from_nsview(&self, ns_view: cocoa::base::id) -> Surface {...}
#[cfg(feature="winit")]
pub fn create_surface(&self, window: &winit::Window) -> Surface {
    use winit::os::macos::WindowExt;
    self.create_surface_from_nsview(WindowExt::get_nsview(window))
}
```